crates/stc_ts_types/src/lib.rs

```diff
@@ -1233,6 +1233,7 @@ pub struct TypeParam {
     pub span: Span,
     pub name: Id,
     pub constraint: Option<Box<Type>>,
+    pub resolved_constraint: Option<Box<Type>>,
```

Comment:
Is this required?

Reply:
Is this for sort of caching?

Reply:
Yep it is for caching. But since it is ok to store in `constraint`, I will just do that instead and remove this.